src/bscPlugin/codeActions/CodeActionsProcessor.ts

```diff
@@ -0,0 +1,108 @@
+import { CodeActionKind } from 'vscode-languageserver';
+import { codeActionUtil } from '../../CodeActionUtil';
+import type { DiagnosticMessageType } from '../../DiagnosticMessages';
+import { DiagnosticCodeMap } from '../../DiagnosticMessages';
+import type { BrsFile } from '../../files/BrsFile';
+import type { XmlFile } from '../../files/XmlFile';
+import type { OnGetCodeActionsEvent } from '../../interfaces';
+import { util } from '../../util';
+
+export class CodeActionsProcessor {
+    public constructor(
+        public event: OnGetCodeActionsEvent
+    ) {
+
+    }
+
+    public process() {
+        for (const diagnostic of this.event.diagnostics) {
+            if (diagnostic.code === DiagnosticCodeMap.callToUnknownFunction) {
+                this.suggestImports(diagnostic as any);
+            } else if (diagnostic.code === DiagnosticCodeMap.xmlComponentMissingExtendsAttribute) {
+                this.addMissingExtends(diagnostic as any);
+            }
+        }
+    }
+
+    private suggestedImports = new Set<string>();
+
+    private suggestImports(diagnostic: DiagnosticMessageType<'callToUnknownFunction'>) {
+        const lowerFunctionName = diagnostic.data.functionName.toLowerCase();
+
+        //skip generating duplicate suggestions if we've already done this one
+        if (this.suggestedImports.has(lowerFunctionName)) {
+            return;
+        }
+
+        this.suggestedImports.add(lowerFunctionName);
+
+        //find the position of the first import statement, or the top of the file if there is none
+        const insertPosition = (this.event.file as BrsFile).parser.references.importStatements[0]?.importToken.range?.start ?? util.createPosition(0, 0);
```

Comment:
Traditionally prefered to add as last import, though could be a problem if imports can be anywhere.

Reply:
yeah, good point. i was just being lazy. I could easily do `importStatements[statements.length-1]` to get the last one. If they have imports anywhere, that's a bsc error, and it's their own fault. :D

Reply:
I fixed this.